src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/timeline/TimelineEventResourceMapper.kt

```diff
@@ -16,6 +17,14 @@ import java.util.UUID
     UUID::class,
   ],
 )
-interface TimelineEventResourceMapper {
-  fun fromDomainToModel(timelineEventDomain: TimelineEvent): TimelineEventResponse
+abstract class TimelineEventResourceMapper {
+  @Mapping(target = "contextualInfo", expression = "java( buildContextualInfo(timelineEventDomain) )")
+  abstract fun fromDomainToModel(timelineEventDomain: TimelineEvent): TimelineEventResponse
+
+  protected fun buildContextualInfo(timelineEventDomain: TimelineEvent): Map<String, String> =
+    if (timelineEventDomain.contextualInfo !== null) {
+      timelineEventDomain.contextualInfo!!.mapKeys { it.key.toString() }
+    } else {
+      emptyMap()
+    }
```

Comment:
Maybe it's a syntax preference thing, but you could use an Elvis operator here instead,

```suggestion
  protected fun buildContextualInfo(timelineEventDomain: TimelineEvent): Map<String, String> =
    timelineEventDomain.contextualInfo?.mapKeys { it.key.toString() } ?: emptyMap()
```

Reply:
Darn it! I've only just seen this comment after merging it!
You are right, that is a better syntax. I forgot about the elvis operator, and started off trying with a ternary which kotlin doesnt support! hence the if/else !
I'll change it to use the elvis operator in my next PR 👍